tests/_internal/concurrency/test_api.py

```diff
@@ -304,34 +304,6 @@ def test_from_sync_call_in_loop_thread(work):
     wait_for_global_loop_exit()
 
 
-@pytest.mark.parametrize("work", [identity, aidentity])
-async def test_from_async_call_in_waiting_thread_from_worker_thread(work):
-    async def worker(parent_thread):
-        result = await from_async.call_in_waiting_thread(
-            create_call(work, 1), parent_thread
-        )
-        assert result == 1
-        return 2
-
-    result = await from_async.wait_for_call_in_new_thread(
-        create_call(worker, threading.current_thread())
-    )
-    assert result == 2
-
-
-@pytest.mark.parametrize("work", [identity, aidentity])
-def test_from_sync_call_in_waiting_thread_from_worker_thread(work):
-    def worker(parent_thread):
-        result = from_sync.call_in_waiting_thread(create_call(work, 1), parent_thread)
-        assert result == 1
-        return 2
-
-    result = from_sync.wait_for_call_in_new_thread(
-        create_call(worker, threading.current_thread())
-    )
-    assert result
-
-
```

Comment:
these were testing the `@staticmethod` from `base` that was never actually used in the library, and this PR inlines `call_in_waiting_thread` to `from_async` and `from_sync`'s versions of `call_soon_in_waiting_thread`